### PR TITLE
RSDK-7203: Remove audio intentions from gostream configuration.

### PR DESCRIPTION
**Pull request by dgottlieb**
There's a windows file too, but I left it untouched.

### CODE REVIEW

**Review by nicksanford — APPROVED** [review]

File: robot/web/web_c.go [hunk 1]
```diff
@@ -106,10 +106,17 @@ func (svc *webService) addNewStreams(ctx context.Context) error {
...
-	newStream := func(name string) (gostream.Stream, bool, error) {
+	newStream := func(name string, isVideo bool) (gostream.Stream, bool, error) {
 		// Configure new stream
-		config := *svc.opts.streamConfig
-		config.Name = name
+		config := gostream.StreamConfig{
+			Name: name,
+		}
+
+		if isVideo {
```
Comment (dgottlieb) [inline robot/web/web_c.go:115]: This code path gets executed when a robot reconfigures and finds a new camera. This was the "bug"/behavior difference that's being changed.

File: robot/web/web_c.go [hunk 2]
```diff
@@ -166,10 +175,11 @@ func (svc *webService) makeStreamServer(ctx context.Context) (*StreamServer, err
 	}
 
 	addStream := func(streams []gostream.Stream, name string, isVideo bool) ([]gostream.Stream, error) {
-		config := *svc.opts.streamConfig
-		config.Name = name
+		config := gostream.StreamConfig{
```
Comment (dgottlieb) [inline robot/web/web_c.go:178]: This was the code path for components that existed on viam-server startup. I changed the behavior from:
* Make a copy of the stream config
* Nil out what we don't want

To instead:
* Make a blank config
* Copy the encoder factories we do* want over

I think the latter flows better and makes this consistent with the code change I had for "new cameras".

File: robot/web/web_c.go [hunk 3]
```diff
@@ -127,7 +134,8 @@ func (svc *webService) addNewStreams(ctx context.Context) error {
 	}
 
 	for name, source := range svc.videoSources {
-		stream, alreadyRegistered, err := newStream(name)
+		const isVideo = true
+		stream, alreadyRegistered, err := newStream(name, isVideo)
```
Comment (dgottlieb) [inline robot/web/web_c.go:138]: This `isVideo` parameter/pattern was copied from how the [startup code works](https://github.com/viamrobotics/rdk/pull/3767/files#diff-db223bcebe63e661c7c3c5b7e0cd20523f08be884e758c78aed2d19ec878f5faR177). I just kept it that way to make these different code paths look as similar as possible.

**Comment by nicksanford** [comment]
This change looks reasonable to me but I'm not going to approve it in it's current state as tests are failing in CI

**Review by nicksanford — APPROVED** [review]

**Pull request merged**